stream-source-kafka/src/main/java/org/apache/kylin/stream/source/kafka/consumer/KafkaConnector.java

```diff
@@ -81,9 +81,15 @@ public void open() {
         } else if (startMode == ConsumerStartMode.LATEST) {
             kafkaConsumer.seekToEnd(topicPartitions);
         } else {
+            List<TopicPartition> newTopicPartitions = Lists.newArrayList();
             for (TopicPartition topicPartition : topicPartitions) {
                 Long offset = partitionOffsets.get(topicPartition.partition());
-                kafkaConsumer.seek(topicPartition, offset);
+                if (offset != null) {
+                    kafkaConsumer.seek(topicPartition, offset);
+                } else {
+                    newTopicPartitions.add(topicPartition);
+                }
+                kafkaConsumer.seekToBeginning(newTopicPartitions);
```

Comment:
I think `kafkaConsumer.seekToBeginning(newTopicPartitions);` should not in **for-loop** .

Reply:
My mistake, thanks.